### PR TITLE
DEV-28331 Config - Deep Merge - array as new array and not merge with old config

### PR DESCRIPTION
**Pull request by sitbubu**
Our problem is that deepmerge library https://www.npmjs.com/package/deepmerge
merges arrays in deepmerge mode by default.

This means that on watchAll we will merge the consul changed array with our current config array,
Instead of running over our current config array with our new consul array.

i.e, 
current config => "maskImpersonationData": {"userIds": [20]}
changed in consul config to => "maskImpersonationData": {"userIds": []}

Will return "maskImpersonationData": {"userIds": [20]} because we deeped merge them, instead of:
"maskImpersonationData": {"userIds": []}"

This is the reason that adding ids work, but deleting them causes an issue of only working after redeploy.

### CODE REVIEW

File: packages/config/src/Config.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ export class Config {
   }
 
   _merge(newValue, onError = false) {
-    const curVales = deepMerge.all([this.#config, newValue]);
+    const overwriteMerge = (destinationArray, sourceArray) => sourceArray;
+    const curVales = deepMerge.all([this.#config, newValue], { arrayMerge: overwriteMerge });
```
[the remark this answers is not in the document]
Comment (sitbubu) [reply packages/config/src/Config.js:30]: Yes, I Will try to see if there is an option to add a flag that defines merge/unmerge dynamically.
If no we will need to think about desired behavior and line up all objects :)

[the remark this answers is not in the document]
Comment (AlonMiz) [reply packages/config/src/Config.js:30]: i think when you WATCH consul and there's a change, you shouldn't merge the new config with the old config.
built it from scratch...
WDYT?

[the remark this answers is not in the document]
Comment (sitbubu) [reply packages/config/src/Config.js]: done

File: packages/config/src/Config.js [hunk 2]
```diff
@@ -65,10 +65,10 @@ export class Config {
     return this.#observables[key].subscribe(onChange);
   }
 
-  set({ value = null, key = null, onError = null }) {
+  set({ defaultValue = null, value = null, key = null, onError = null }) {
```
Comment (AlonMiz) [inline packages/config/src/Config.js:68]: what does defaultValue stands for?

Comment (AlonMiz) [reply packages/config/src/Config.js:68]: old value?

File: packages/config/src/Config.js [hunk 3]
```diff
@@ -25,8 +25,8 @@ export class Config {
     }
   }
 
-  _merge(newValue, onError = false) {
-    const curVales = deepMerge.all([this.#config, newValue]);
+  _merge({ defaultValue = null, newValue, onError = false }) {
+    const curVales = deepMerge.all([defaultValue || this.#config, newValue]);
```
Comment (AlonMiz) [inline packages/config/src/Config.js:29]: im not sure what's going on here.
also a test on that matter would be better to help us understand what's happening and when

File: packages/config/src/Config.js (hunk 2 above)
Comment (sitbubu) [reply packages/config/src/Config.js:68]: It represents whatever we like, e.g:
https://github.com/logzio/gaia-hermes-ws/pull/8373
We passed our original config as default in order to merge our newValue to the originalValue on watchAll.

**Comment by sitbubu** [comment]
@AlonMiz ping

File: packages/config/__test__/Config.test.js [hunk 4]
```diff
@@ -114,6 +114,24 @@ describe('Config - create config with schema', () => {
     expect(config.get('teams.name')).toEqual('override name2');
   });
 
+  it('should deepMerge initial scheme default array with new array', () => {
+    const schema = Joi.object({
+      defaultIdArr: Joi.array().items(Joi.number()).default([20]),
```
Comment (AlonMiz) [inline packages/config/__test__/Config.test.js:119]:
```suggestion
      ids: Joi.array().items(Joi.number()).default([20]),
```

File: packages/config/src/Config.js [hunk 5]
```diff
@@ -65,8 +69,9 @@ export class Config {
     return this.#observables[key].subscribe(onChange);
   }
 
-  set({ value = null, key = null, onError = null }) {
+  set({ cleanMerge = false, value = null, key = null, onError = null }) {
```
Comment (AlonMiz) [inline packages/config/src/Config.js:72]:
```suggestion
  set({ override = false, value = null, key = null, onError = null }) {
```

Comment (AlonMiz) [reply packages/config/src/Config.js:72]: or
```suggestion
  set({ merge = true, value = null, key = null, onError = null }) {
```

File: packages/config/__test__/Config.test.js [hunk 6]
```diff
@@ -114,6 +114,24 @@ describe('Config - create config with schema', () => {
...
+    expect(config.get('defaultIdArr')).toEqual([20]);
+    config.set({ value: { defaultIdArr: [22] } });
+    expect(config.get('defaultIdArr')).toEqual([20, 22]);
+    config.set({ cleanMerge: true, value: { defaultIdArr: [22, 23] } });
+    expect(config.get('defaultIdArr')).toEqual([20, 22, 23]);
+    config.set({ cleanMerge: true, value: { defaultIdArr: [24] } });
+    expect(config.get('defaultIdArr')).toEqual([20, 24]);
+    config.set({ cleanMerge: true, value: { defaultIdArr: [] } });
+    expect(config.get('defaultIdArr')).toEqual([20]);
+  });
```
Comment (AlonMiz) [inline packages/config/__test__/Config.test.js:133]: amazing

**Review by AlonMiz — APPROVED** [review]
AMazing

**Pull request merged**